ios/FluentUI/Card Nudge/CardNudgeTokens.swift

```diff
@@ -15,97 +15,122 @@ import SwiftUI
+                              buttonBackgroundColor: ColorSet? = nil,
+                              buttonInnerPaddingHorizontal: CGFloat? = nil,
+                              circleRadius: CGFloat? = nil,
+                              circleSize: CGFloat? = nil,
+                              cornerRadius: CGFloat? = nil,
+                              horizontalPadding: CGFloat? = nil,
+                              iconSize: CGFloat? = nil,
+                              interTextVerticalPadding: CGFloat? = nil,
+                              mainContentVerticalPadding: CGFloat? = nil,
+                              minimumHeight: CGFloat? = nil,
+                              outlineColor: ColorSet? = nil,
+                              outlineWidth: CGFloat? = nil,
+                              subtitleTextColor: ColorSet? = nil,
+                              textColor: ColorSet? = nil,
+                              verticalPadding: CGFloat? = nil) -> CardNudgeTokens {
+
+        let tokens: CardNudgeTokens
         switch style {
         case .standard:
-            return aliasTokens.backgroundColors[.neutral2]
+            tokens = CardNudgeTokens()
         case .outline:
-            return aliasTokens.backgroundColors[.neutral1]
+            tokens = OutlineCardNudgeTokens()
         }
-    }
 
-    open var buttonBackgroundColor: ColorSet {
-        return globalTokens.brandColors[.tint30]
-    }
+        // Optional overrides
+        tokens._accentColorOverride = accentColor
+        tokens._accentIconSizeOverride = accentIconSize
+        tokens._accentPaddingOverride = accentPadding
+        tokens._backgroundColorOverride = backgroundColor
+        tokens._buttonBackgroundColorOverride = buttonBackgroundColor
+        tokens._buttonInnerPaddingHorizontalOverride = buttonInnerPaddingHorizontal
+        tokens._circleRadiusOverride = circleRadius
+        tokens._circleSizeOverride = circleSize
+        tokens._cornerRadiusOverride = cornerRadius
+        tokens._horizontalPaddingOverride = horizontalPadding
+        tokens._iconSizeOverride = iconSize
+        tokens._interTextVerticalPaddingOverride = interTextVerticalPadding
+        tokens._mainContentVerticalPaddingOverride = mainContentVerticalPadding
+        tokens._minimumHeightOverride = minimumHeight
+        tokens._outlineColorOverride = outlineColor
+        tokens._outlineWidthOverride = outlineWidth
+        tokens._subtitleTextColorOverride = subtitleTextColor
+        tokens._textColorOverride = textColor
+        tokens._verticalPaddingOverride = verticalPadding
 
-    open var buttonInnerPaddingHorizontal: CGFloat {
-        return globalTokens.spacing[.small]
+        return tokens
     }
 
-    open var circleRadius: CGFloat {
-        return globalTokens.borderRadius[.circle]
-    }
+    fileprivate var _accentColorOverride: ColorSet?
+    var accentColor: ColorSet { _accentColorOverride ?? globalTokens.brandColors[.shade20] }
```

Comment:
I'm not a big fan of having separate override storage from the actual token getter, but I'm at a loss as to how to support all three of the following features:
- Lazy default values that reference the superclass's `globalTokens` and `aliasTokens` properties.
- Internal overrides for alternate "styles" (e.g. `.outline` vs `.standard` here), which _also_ lazily reference `[global|alias]Tokens`.
- Optional storage of custom properties, of a variety of different types for different tokens (which makes flat array/dictionary storage hard).

If anyone has any ideas, I'm eager to hear them!

Reply:
I'm still working to understand the changes, but would it be possible to just treat our different styles the same way we treat external custom tokens? Do we already do that? Seems like different styles and custom tokens should be fairly similar. Or do the custom tokens also make those three features hard to satisfy?

Reply:
Great question! That's basically what last week's version (ea1a2bcecdaebb0d9a854ecfcec9d7bf8fea0cd9) tried, but it ran into a problem with custom global tokens. 

See, the tricky part is that custom tokens are constant values, while our styles can be dynamically pulled from the global or alias token sets, e.g. for brand colors. It's easy to override a property with `Color.red`, but I haven't been able to think of a way to override it with something like `self.globalTokens.brandColors[.primary]`, particularly since `globalTokens` won't yet be set at `init()`.